fuel-p2p/src/gossipsub/builder.rs

```diff
@@ -0,0 +1,65 @@
+use libp2p::{
+    gossipsub::{
+        FastMessageId, Gossipsub, GossipsubConfigBuilder, GossipsubMessage, MessageAuthenticity,
+        MessageId, PeerScoreParams, PeerScoreThresholds, RawGossipsubMessage, ValidationMode,
+    },
+    identity::Keypair,
+};
+use sha2::{Digest, Sha256};
+use std::time::Duration;
+
+pub fn build_gossipsub(local_key: &Keypair) -> Gossipsub {
+    let gossip_message_id =
+        move |message: &GossipsubMessage| MessageId::from(&Sha256::digest(&message.data)[..20]);
+
+    let fast_gossip_message_id = move |message: &RawGossipsubMessage| {
+        FastMessageId::from(&Sha256::digest(&message.data)[..8])
+    };
+
+    let gossipsub_config = GossipsubConfigBuilder::default()
```

Comment:
These are a lot of options 😂

1. Is there some sort of rationale for the parameter choice?
2. Is there a better way to do this?
3. If there's no functional ordering requirement, can these to ordered alphabetically?

Reply:
in short, we probably won't need many of those, so they will be removed in final code (they are set by default) 

those that we need, we can order them nicely, that was completely arbitrary

Reply:
We should have some kind of network configuration object we can pass into this `build_gossipsub` method. Then we can put all these defaults there and override them using cli args or service config later.

Reply:
yes they will be part of the general p2p config, I will update it